File: src/main/java/pw/switchcraft/plethora/gameplay/modules/sensor/SensorModuleItem.java
package pw.switchcraft.plethora.gameplay.modules.sensor;

import net.minecraft.util.Identifier;
import pw.switchcraft.plethora.gameplay.modules.LevelableModuleItem;
import pw.switchcraft.plethora.util.config.Config.Sensor;

import javax.annotation.Nonnull;

import static pw.switchcraft.plethora.Plethora.MOD_ID;

public class SensorModuleItem extends LevelableModuleItem {
    public static final Identifier MODULE_ID = new Identifier(MOD_ID, "sensor");

    public SensorModuleItem(Settings settings) {
        super("sensor", settings);
    }

    @Override
    public int getBaseRange() {
        return Sensor.radius;
    }

    @Override
    public int getMaxRange() {
        return Sensor.maxRadius;
    }

    @Override
    public int getLevelCost() {
        return Sensor.senseLevelCost;
    }

    @Nonnull
    @Override
    public Identifier getModule() {
        return MODULE_ID;
    }
}

File: src/main/java/pw/switchcraft/plethora/gameplay/modules/sensor/SensorModuleItem.kt
package pw.switchcraft.plethora.gameplay.modules.sensor

import net.minecraft.util.Identifier
import pw.switchcraft.plethora.Plethora
import pw.switchcraft.plethora.gameplay.modules.LevelableModuleItem
import pw.switchcraft.plethora.gameplay.registry.PlethoraModules

class SensorModuleItem(settings: Settings) : LevelableModuleItem("sensor", settings) {
  private val cfg by Plethora.config::sensor

  override fun getModule(): Identifier = PlethoraModules.SENSOR_M

  override fun getBaseRange() = cfg.radius
  override fun getMaxRange() = cfg.maxRadius
  override fun getLevelCost() = cfg.senseLevelCost
}
